tests/autoyast/prepare_profile.pm

```diff
@@ -31,6 +32,24 @@ sub run {
     # Get file from data directory
     my $profile = get_test_data($path);
 
+    # Add registration block for cloned autoyast profiles due to
+    # the registration code is never clonned as it is not stored in the system anywhere
+    if (get_var('AUTOYAST_ADD_REGISTRATION')) {
+        # Create registration block
+        my $registration = << "EOF";
+  <suse_register>
+    <do_registration config:type="boolean">true</do_registration>
+    <email/>
+    <reg_code>{{SCC_REGCODE}}</reg_code>
+    <install_updates config:type="boolean">true</install_updates>
+    <reg_server>{{SCC_URL}}</reg_server>
+  </suse_register>
+  <add-on>
+EOF
+        # Inject registration block
+        $profile =~ s/<add-on>/$registration/;
```

Comment:
You can use capture groups to replace `<add-on>` with `<add-on>`+something.
Not sure about exact syntax, but something like s/(<add-on>)/\$1$registration/. You can also use named capture groups, like 
```
$variable =~ /(?<count>\d+)/;
print "Count is $+{count}";
```

Reply:
cool, I didn't know this magic could be also useful for replacement, thanks for tip. Added.